strictdoc/backend/sdoc/models/document_bibliography.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional, List
+
+from pybtex.database import BibliographyData, parse_file
+
+from strictdoc.backend.sdoc.models.type_system import (
+    BibEntry,
+    FileEntry,
+    FileEntryFormat,
+)
+
+
+class DocumentBibliography:
+    def __init__(
+        self,
+        parent,
+        bib_files: Optional[List[FileEntry]],
+        bib_entries: Optional[List[BibEntry]],
+    ):
+        self.parent = parent
+        self.bib_db = BibliographyData()
+
+        self.bib_files = bib_files
+        self.bib_entries = bib_entries
+
+        if bib_files and len(bib_files) > 0:
+            for bib_file in bib_files:
+                if bib_file.file_format == FileEntryFormat.BIBTEX:
```

Comment:
And another pedantic comment: Is the expectation here that we always have the `FileEntryFormat.BIBTEX` type? How about adding an assert just above and left-indenting the whole code block by removing the whole if condition? Or you expect some variability here in the future? The motive is to make it strict and catch strange behavior as early as possible.